fluentd/vendored_gem_src/syslog_protocol/lib/syslog_protocol/packet.rb

```diff
@@ -23,6 +40,15 @@ def assemble(max_size = 1024)
       data
     end
 
+    def assemble_rfc5424()
+      unless @hostname and @facility and @severity and @appname
```

Comment:
I believe you want `&&` not `and` https://mixandgo.com/learn/understanding-boolean-operator-precedence-in-ruby

Reply:
updated